core/frontend/src/utils/pull_tracker.ts

```diff
@@ -145,5 +145,28 @@ class PullTracker {
 
     this.updateSimplifiedProgress()
   }
+
+  digestNewData(progressEvent: {currentTarget: { response: string }}, parseFragments = true): void {
+    let buffer = progressEvent?.currentTarget?.response
+
+    if (parseFragments) {
+      const result = aggregateStreamingResponse(
+        parseStreamingResponse(progressEvent?.currentTarget?.response),
+        (fragment) => {
+          this.onerror(fragment.error ?? `Unknown error with status ${fragment.status}`)
+          /** Stops aggregation */
+          return false
+        },
+      )
+
+      if (!result) {
```

Comment:
An empty string will also be a valid value here, is this correct ?

Reply:
Changed to check only for undefined.